HW04.Rmd

```diff
@@ -69,51 +129,146 @@ Assume that:
+    scale_y_continuous(breaks=seq(0, 60000, by = 10000)) + 
+    theme(axis.text.x = element_text(angle = 45, hjust = 1))
 
 ```
 
 (hint maybe looking at percentages would be good)
 ```{r race_2}
-
+# summarize percentage of intent within each race group, limit race groups to White, Black, and Hispanic
+gun_deaths %>%
+    group_by(race, intent) %>%
+    count() %>%
+    group_by(race) %>%
+    mutate(percent = n / sum(n) ) %>%
+    select(race, intent, percent) %>%
+    filter(intent %in% c("Suicide", "Homicide") & race %in% c("White", "Black", "Hispanic")) 
 ```
 
 
 ### Are police-involved gun deaths significantly different from other gun deaths? Assess the relationship between police involvement and age, police involvement and race, and the intersection of all three variables.
 
 ```{r police}
-
+# convert police-involved from numeric to factor
+gun_deaths$police <- factor(as.character(gun_deaths$police))
+summary(gun_deaths$police)
+
+# select variables needed for analysis and remove missng values
+police <- gun_deaths %>%
+    select(police, age, race) %>%
+    filter(is.na(age) == FALSE) 
+
+# calculate average age by police-involved
+avg_age_police <- police %>%
+    group_by(police) %>%
+    summarize(mean_age=mean(age, na.rm = TRUE)) 
+
+# round to the nearest tenth
+avg_age_police$mean_age <- round(avg_age_police$mean_age, digits = 2)
+
+# draw box plot of age by police-involved
+ggplot(police, aes(police, age)) +
+    geom_boxplot(color="blue") +
+    xlab("\nPolice-involved") + 
+    ylab("Age\n") +
+    scale_y_continuous(breaks=seq(0, 100, by = 10)) + 
+    stat_summary(fun = mean, colour = "red", geom = "point", shape = 18, size = 2) +
+    geom_text_repel(avg_age_police, mapping = aes(x = police, y = mean_age, label = mean_age), inherit.aes = FALSE) + 
+    scale_x_discrete(breaks=c("0", "1"), labels=c("Uninvolved", "Involved"))
 ```
 
 Think about performing stats such as a t.test to see if differences you are plotting are statistically significant
 
 ```{r stats?}
-
-
+# use t-test to check if the differences between two means are statistically significant
+t.test(police$age ~ police$police)
 ```
 
+**The results of t-test suggest that the differences are statistically significant
 
 ```{r police_2}
-
+# create a new data set with percentage of police-involved death within each race group
+police_2 <- police %>%
+    group_by(race, police) %>%
+    count() %>%
+    ungroup() %>%
+    group_by(race) %>%
+    mutate(percent = n / sum(n) * 100) %>%
+    ungroup()
+
+# round percent to the tenth
+police_2$percent = round(police_2$percent, digit = 2)
+
+# draw a bar chart to visualize the differences across race groups
+ggplot(police_2, aes(race, percent, fill = police)) + 
+    geom_col(position = "dodge") + 
+    xlab("\nRace") +
+    ylab("Percent of Police-involved Death\n") + 
+    scale_y_continuous(breaks=seq(0, 100, by = 10)) + 
+    theme(axis.text.x = element_text(angle = 45, hjust = 1))

```

Comment:
How about try stacked bar plots such that it is easier to compare the percentages?

Reply:
Great advice! Thanks!